### PR TITLE
Adding accuracy checks to integration tests

### PR DESCRIPTION
**Pull request by oberqi**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs change / refactoring / dependency upgrade

## Motivation and Context / Related issue

We currently have a set of integration tests run by CircleCI - we run our examples (MNIST, CIFAR, etc trained with Opacus) and check if they're completed successfully.

What we don't check is that the training outcome make sense - we only check the exit code. 
There are few things that could go wrong without us noticing, for example if we break noise calibration.

We want to look at the final accuracy too and fail if something is off - to catch any unexpected regressions,

## How Has This Been Tested (if it applies)

Run unit and integration tests

## Checklist

- [X ] The documentation is up-to-date with the changes I made.
- [X ] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [X ] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -158,6 +159,7 @@ commands:
             echo "Using $(pip -V) ($(which pip))"
             pip install tensorboard
             python examples/cifar10.py --lr 0.25 --sigma 0.7 -c 1.5 --sample-rate 0.001 --epochs 1 --data-root runs/cifar10/data --log-dir runs/cifar10/logs --device <<parameters.device>>
+            python -c "import torch; model = torch.load('model_best.pth.tar'); exit(0) if (model['best_acc1']>0.095 and model['best_acc1']<0.12) else exit(1)"
```
Comment (ffuuugor) [inline .circleci/config.yml:162]: This doesn't look right - 0.1 accuracy on CIFAR10 is essentially random, so we're not really getting any signal here. We should probably run for few more epochs. We don't need to get to high accuracy (it'll probably take few dozen epochs and doesn't worth it), but something reliably different from random would help. It only takes a few seconds per epoch, so shouldn't be a problem

File: .circleci/config.yml [hunk 2]
```diff
@@ -201,6 +203,7 @@ commands:
             echo "Using $(pip -V) ($(which pip))"
             pip install --user datasets transformers
             python examples/imdb.py --lr 0.02 --sigma 0.56 -c 1.0 --sample-rate 0.00256 --max-sequence-length 256 --epochs 1 --data-root runs/imdb/data --device <<parameters.device>>
+            python -c "import torch; model = torch.load('run_results_imdb_classification.pt'); exit(0) if (model>0.54 and model<0.66) else exit(1)"
```
Comment (ffuuugor) [inline .circleci/config.yml:206]: nit: rename model -> accuracy or smth

File: examples/imdb.py [hunk 3]
```diff
@@ -288,7 +290,8 @@ def main():
 
     for epoch in range(1, args.epochs + 1):
         train(args, model, train_loader, optimizer, epoch)
-        evaluate(args, model, test_loader)
+        mean_accuracy = evaluate(args, model, test_loader)
+        torch.save(mean_accuracy, "run_results_imdb_classification.pt")
```
Comment (ffuuugor) [inline examples/imdb.py:294]: nit: probably no point in saving it after each epoch?

**Review by ffuuugor — COMMENTED** [review]
Looks great, thanks for the clean and simple solution!

The only thing I'm not accepting yet is CIFAR10 test - with the current setup we're not actually testing anything meaningful. 
The rest comments are nits, feel free not to focus on them.

NB: If you're having problems with isort from CircleCI linter, make sure to rebase past [this commit](https://github.com/pytorch/opacus/commit/6e914d93ad3dc1cd480a46bdf9d530b55aa4cb5e)

File: examples/imdb.py (hunk 3 above)
Comment (oberqi) [reply examples/imdb.py:294]: thanks for spotting!

File: .circleci/config.yml (hunk 1 above)
Comment (oberqi) [reply .circleci/config.yml:162]: Accuracy remains 0.1 after 10 epochs, creating an issue for that

**Comment by ffuuugor** [comment]
Thanks for the updates! 
Looking at the [CircleCI tests](https://github.com/pytorch/opacus/pull/210/checks?check_run_id=3723214928) for the latest commit  the linter still complains about `isort` errors. Can you pls double check you've pulled the latest changes? (isort fixed [here](https://github.com/pytorch/opacus/commit/6e914d93ad3dc1cd480a46bdf9d530b55aa4cb5e))

Also don't forget to re-import the PR to Phabricator when you're done

**Comment by oberqi** [comment]
> Thanks for the updates!
> Looking at the [CircleCI tests](https://github.com/pytorch/opacus/pull/210/checks?check_run_id=3723214928) for the latest commit the linter still complains about `isort` errors. Can you pls double check you've pulled the latest changes? (isort fixed [here](https://github.com/pytorch/opacus/commit/6e914d93ad3dc1cd480a46bdf9d530b55aa4cb5e))
> 
> Also don't forget to re-import the PR to Phabricator when you're done

Yes I have been building on top of that commit, not sure what I am missing here
image

**Comment by ffuuugor** [comment]
Alright, so this is partially my fault, but I still don't fully understand what's going on.

tl;dr - can you pls check if your local isort is the same version as on CircleCI servers (5.9.3), run and submit with changes it proposes (including files you didn't touch)

# Long story
In a somewhat surprising fashion, different versions of isort produce different results on the same file (devservers don't use `pip` so there's a mismatch: 4.3.21 on the devserver vs 5.9.3 on CircleCI). Combine that with our lack of custom configuration and you get inconsistent results.

To make things worse, github/phabricator syncing failed on my diff, which was supposed to fix the issue. I naively thought, that isort commit couldn't go wrong and landed it without properly running CircleCI - which, as we now know, was a mistake.

**Comment by oberqi** [comment]
> Alright, so this is partially my fault, but I still don't fully understand what's going on.
> 
> tl;dr - can you pls check if your local isort is the same version as on CircleCI servers (5.9.3), run and submit with changes it proposes (including files you didn't touch)
> 
> # Long story
> In a somewhat surprising fashion, different versions of isort produce different results on the same file (devservers don't use `pip` so there's a mismatch: 4.3.21 on the devserver vs 5.9.3 on CircleCI). Combine that with our lack of custom configuration and you get inconsistent results.
> 
> To make things worse, github/phabricator syncing failed on my diff, which was supposed to fix the issue. I naively thought, that isort commit couldn't go wrong and landed it without properly running CircleCI - which, as we now know, was a mistake.

I have 5.9.3, I have pushed with proposed changes (it suggested changes on 42 files)
